python_ta/cfg/visitor.py

```diff
@@ -437,6 +439,64 @@ def visit_with(self, node: nodes.With) -> None:
         for child in node.body:
             child.accept(self)
 
+    def visit_match(self, node: nodes.Match) -> None:
+        """Visit a match statement and create appropriate control flow."""
+        # When only creating cfgs for functions, _current_cfg will only be None outside of functions
+        if self._current_cfg is None:
+            return
+
+        self._current_block.add_statement(node.subject)
+        node.cfg_block = self._current_block
+        after_match_block = self._current_cfg.create_block()
+
+        case_end_blocks = []
+
+        prev_case = self._current_block
+        # Process each match case
+        for case in node.cases:
+
+            edge_label = "No Match" if case_end_blocks else ""
+
+            new_case = astroid.nodes.Name(
+                name=f"Case {case.pattern.as_string()}",
+                lineno=case.lineno,
+                col_offset=case.col_offset,
+                parent=case.parent,
+                end_col_offset=case.end_col_offset,
+                end_lineno=case.end_lineno,
+            )
+
+            separate_conditions = self.options.get("separate-condition-blocks", False)
+            if separate_conditions and hasattr(case, "guard") and case.guard is not None:
+                # If the option is set to separate condition blocks, create a new block for the guard
+                pattern_block = self._current_cfg.create_block(prev_case)
+                pattern_block.add_statement(case.guard)
+                pattern_block = self._current_cfg.create_block(pattern_block, edge_label=edge_label)
+            else:
+                pattern_block = self._current_cfg.create_block(prev_case, edge_label=edge_label)
+
+            pattern_block.add_statement(new_case)
+            pattern_body = self._current_cfg.create_block(pattern_block, edge_label="Match")
+
+            self._current_block = pattern_body
+
+            if not separate_conditions and hasattr(case, "guard") and case.guard is not None:
```

Comment:
This logic isn't correct. The guard should _always_ be separated from the pattern body, as the body is only executed if the guard is `True`. In that sense, the structure should mimic that of an if statement (labels with `True` and `False` leaving the guard; note that the `False` branch should lead to the next case).

Really what the `separate-condition-blocks` option can do is trigger whether the guard appears in the same block *as the case itself*, not whether it appears in the same block as the pattern. In the case where the guard is combined with the case, you can again customize the rendering of it in the `_visit` function (see above comment).